### ISSUE TITLE
Duplicate log_softmax function in decoder output and loss calculation

### ISSUE THREAD

**Issue by zhongpeixiang** [spro/practical-pytorch#90]
For seq2seq-translation-batched tutorial,

In the `forward `function of `BahdanauAttnDecoderRNN`, It has a log_softmax function on the output:

`output = F.log_softmax(self.out(torch.cat((output, context), 1)))`

In the **masked_cross_entropy.py** file, it has another log_softmax function on the decoder output before calculating cross entropy loss:

`log_probs_flat = functional.log_softmax(logits_flat)`

Which I think are duplicates. and only one log_softmax is needed to calculate cross entropy loss
